dotCMS/src/integration-test/java/com/dotcms/storage/StoragePersistenceAPITest.java

```diff
@@ -0,0 +1,347 @@
+    }
+
+    /**
+     * Given scenario: We want corroborate that we're able to recover files and verify the existence of a group regardless of casing.
+     * Expected Results: Regardless of casing we should be able to retrieve and find objects
+     * @param testCase
+     */
+    @Test
+    @UseDataProvider("getRandomTestCases")
+    public void Test_Pull_File_Different_Casing(final TestCase testCase) throws DotDataException {
+        final StoragePersistenceAPI storage = persistenceProvider.getStorage(testCase.storageType);
+        if(!storage.existsGroup(testCase.groupName)){
+           assertTrue(storage.createGroup(testCase.groupName));
+        }
+        final Object object = storage
+                .pushFile(testCase.groupName, testCase.path, testCase.file, ImmutableMap.of());
+        assertNotNull(object);
+
+        final String randomCaseGroupName = upperCaseRandom(testCase.groupName, testCase.groupName.length());
+        assertTrue(storage.existsGroup(randomCaseGroupName));
+        final String randomCasePath = upperCaseRandom(testCase.path, testCase.path.length());
+        final File file = storage.pullFile(randomCaseGroupName, randomCasePath);
+        assertNotNull(file);
+
+    }
+
+    @DataProvider
+    public static Object[] getRandomTestCases() throws Exception{
+        final String path = "any-path";
+        final String groupName = RandomStringUtils.randomAlphanumeric(10);
+        final File temp = File.createTempFile("tempfile", ".txt");
+        return new Object[]{
+                new TestCase(StorageType.FILE_SYSTEM, groupName, path, temp),
+                new TestCase(StorageType.DB, groupName, path, temp)
+        };
+    }
+
+    /**
+     * Given scenario:
+     * Expected Results:
+     * @param testCase
+     * @throws IOException
+     * @throws NoSuchAlgorithmException
+     */
+    @Test
+    @UseDataProvider("getLargeFileTestCases")
+    public void Test_Push_Large_File(final TestCase testCase)
+            throws IOException, NoSuchAlgorithmException, DotDataException {
+
+        final StoragePersistenceAPI storage = persistenceProvider.getStorage(testCase.storageType);
+        if (!storage.existsGroup(testCase.groupName)) {
+            assertTrue(storage.createGroup(testCase.groupName));
+        }
+        storage.pushFile(testCase.groupName, testCase.path, testCase.file, ImmutableMap.of());
+
+        final File pullFile = storage.pullFile(testCase.groupName, testCase.path);
+        assertEquals(testCase.file.length(), pullFile.length());
+
+        assertEquals(testCase.file.length(), LARGE_FILE_SIZE);
+
+        assertEquals(Encryptor.Hashing.sha256().append(Files.readAllBytes(testCase.file.toPath())).buildUnixHash(),
+                     Encryptor.Hashing.sha256().append(Files.readAllBytes(pullFile.toPath())).buildUnixHash());
+
+    }
+
+    private static final Random random = new Random();
+
+    @DataProvider
+    public static Object[] getLargeFileTestCases() throws Exception{
```

Comment:
Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)